javacord-core/src/main/java/org/javacord/core/entity/server/ServerUpdaterDelegateImpl.java

```diff
@@ -516,31 +516,31 @@ public void reorderRoles(List<Role> roles) {
 
     @Override
     public void addRoleToUser(User user, Role role) {
-        Collection<Role> userRoles = this.userRoles.computeIfAbsent(user, u -> new ArrayList<>(server.getRoles(u)));
+        Collection<Role> userRoles = this.userRoles.computeIfAbsent(user, u -> new HashSet<>(server.getRoles(u)));
```

Comment:
```suggestion
        Set<Role> userRoles = this.userRoles.computeIfAbsent(user, u -> new HashSet<>(server.getRoles(u)));
```
^ Same for the methods below

The userRoles variable declaration can also be changed to 
`private final Map<User, Set<Role>> userRoles`